packages/custom-functions-metadata/src/parseTree.ts

```diff
@@ -1029,50 +1055,84 @@ function hasInvocationParameter(
 
 /**
  * Gets the parameter type of the node
- * @param t TypeNode
+ * @param node TypeNode
  */
-function getParamType(t: ts.TypeNode, extra: IFunctionExtras, enumList: string[]): string {
+function getParamType(node: ts.TypeNode, extra: IFunctionExtras, enumList: string[]): string {
   let type = "any";
-  // Only get type for typescript files.  js files will return any for all types
-  if (t) {
-    let kind = t.kind;
-    const typePosition = getPosition(t);
-    if (ts.isTypeReferenceNode(t) || ts.isArrayTypeNode(t)) {
-      let arrayType: IArrayType = {
-        dimensionality: 0,
-        type: ts.SyntaxKind.AnyKeyword,
-      };
-      if (ts.isTypeReferenceNode(t)) {
-        const array = t as ts.TypeReferenceNode;
-        if (enumList.indexOf(array.typeName.getText()) >= 0) {
-          // Type found in the enumList
-          return type;
-        }
-        if (array.typeName.getText() !== "Array") {
-          extra.errors.push(logError("Invalid type: " + array.typeName.getText(), typePosition));
-          return type;
-        }
-      }
-      arrayType = getArrayDimensionalityAndType(t);
-      kind = arrayType.type;
+  // Only get type for typescript files. js files will return "any" for all types
```

Comment:
When using JavaScript, the user can annotate the type in the @param field, so it should be able to have a type when annotated for JavaScript, and we would expect this to be done for JavaScript in order to know the type.

Perhaps you want to add an error unit test case when a param in JavaScript doesn't specify a type, or perhaps this is the same any actually specifying `any` or `unknown` as the type in TypeScript. Might be useful to have test cases to clarify this.

In terms of the code here, it seems that not having a node implies that the type to return is any, so the code looks fine.

Reply:
Resolving as no issue present.